framework/src/ble_manager/ble_manager_state.c

```diff
@@ -261,6 +263,72 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
 		}
 	} break;
 
+	case BLE_EVT_CMD_CLIENT_ENABLE_AUTOCONNECT: {
+		ble_client_ctx *ctx = (ble_client_ctx *)msg->param;
+
+		if (ctx == NULL) {
+			ret = TRBLE_INVALID_ARGS;
+			break;
+		}
+
+		if (ctx->state != BLE_CLIENT_IDLE && ctx->state != BLE_CLIENT_CONNECTED) {
+			ret = TRBLE_INVALID_STATE;
+			break;
+		}
+
+		ctx->auto_connect = true;
+		ret = TRBLE_SUCCESS;
+	} break;
+
+	case BLE_EVT_CMD_CLIENT_DISABLE_AUTOCONNECT: {
+		ble_client_ctx *ctx = (ble_client_ctx *)msg->param;
+		
+		if (ctx == NULL || ctx->send_mqfd <= 0) {
+			ret = TRBLE_INVALID_ARGS;
+			break;
+		}
+		
+		if (ctx->state != BLE_CLIENT_AUTOCONNECTING) {
+			ret = TRBLE_INVALID_STATE;
+			break;
+		}
+
+		char buf[BLE_MQ_SIZE] = { 0, };
+		buf[0] = BLE_AUTOCON_EVT_ALERT;
+		int status = mq_send(ctx->send_mqfd, buf, BLE_MQ_SIZE, 0);
+		if (status < 0) {
+			BLE_LOG_ERROR("[BLEMGR] fail to send mqueue[%d]\n", status);
+			ret = TRBLE_FILE_ERROR;
+			break;
+		}
+		ret = TRBLE_SUCCESS;
+	} break;
+
+	case BLE_EVT_CMD_CLIENT_RECONNECT: {
+		BLE_STATE_CHECK;
+
+		ble_client_ctx *ctx = (ble_client_ctx *)msg->param;
+
+		if (ctx == NULL) {
+			ret = TRBLE_INVALID_ARGS;
+			break;
+		}
+
+		if (ctx->state == BLE_CLIENT_IDLE) {
+			ctx->state = BLE_CLIENT_CONNECTING;
+		} else if (ctx->state == BLE_CLIENT_AUTOCONNECTING) {
```

Comment:
If this condition is not needed, please remove this.

Reply:
Ok, I will fix it